python/ray/serve/_private/proxy.py

```diff
@@ -1198,13 +1198,27 @@ def __init__(
             proxy_router=self.proxy_router,
             request_timeout_s=self._http_options.request_timeout_s,
         )
+
+        if (
+            RAY_SERVE_REQUEST_PROCESSING_TIMEOUT_S is not None
+            and self._grpc_options.request_timeout_s is None
+        ):
+            self._grpc_options.request_timeout_s = (
+                RAY_SERVE_REQUEST_PROCESSING_TIMEOUT_S
+            )
+            logger.warning(
+                "The `SERVE_REQUEST_PROCESSING_TIMEOUT_S` environment variable will be deprecated "
+                "in the future release. Please set `request_timeout_s` in your Serve config's `grpc_options` "
+                "field instead."
+            )
+
```

Comment:
nit: move this into a new function called `_set_proxy_default_grpc_options` similar to how its done for http. This will help reduce some code clutter in this function.